File: Fcfs.java
import java.util.*;
public class Fcfs
{
    public static void main(String args[])
    {
        int i,twt=0,ttat=0;
        int[] wt = new int[10];
        int[] bt = new int[10];
        int[] tat = new int[10];
        
        float avgwt,avgtat;
        System.out.println("Enter How Many Processes?");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        for(i=0;i<n;i++)
        {
            System.out.println("Enter Burst Time for process:"+(i+1));
            bt[i]=sc.nextInt();
        }
        wt[0] =0;
        for(i=0;i<n;i++)
        {
            wt[i+1]=wt[i]+bt[i];
            tat[i]=wt[i]+bt[i];
            twt=twt+wt[i];
            ttat=ttat+tat[i];      
        }
            avgwt=(float)twt/n;
            avgtat=(float)ttat/n;
            System.out.println("\nProcessID  BurstTime  WaitingTime  TurnAroundTime\n");
            for(i=0;i<n;i++)
            {
                System.out.println((n+1)+"\t"+bt[i]+"\t"+wt[i]+"\t"+tat[i]);
            }
                System.out.println("\nTotal Waiting Time"+twt+"\n");
                 System.out.println("Total TurnAroundTime:"+ttat+"\n\n");
                 System.out.println("Average Waiting Time:"+avgwt+"\n");
                 System.out.println("Average TurnAroundTime:"+avgtat+"\n");
                
            
           
    }
}

File: fcfs.c
#include<stdio.h>

int main()
{
 int i,n,wt[10],bt[10],tat[10],twt=0,ttat=0;
 float avgwt,avgtat;

 printf("Enter How Many Processes?");
 scanf("%d",&n);
 for(i=0;i<n;i++)
 {
  printf("Enter Burst Time for process %d:",i+1);
  scanf("%d",&bt[i]);
 }
 wt[0]=0;
 for(i=0;i<n;i++)
 {
  wt[i+1]=wt[i]+bt[i];
  tat[i]=wt[i]+bt[i];
  twt=twt+wt[i];
  ttat=ttat+tat[i];
 }
 avgwt=(float)twt/n;
 avgtat=(float)ttat/n;
 printf("\nProcessID  BurstTime  WaitingTime  TurnAroundTime\n");
 for(i=0;i<n;i++)
  printf("\n\n%d\t\t%d\t\t%d\t\t%d\n",n+1,bt[i],wt[i],tat[i]);
 printf("\nTotal Waiting Time:%d\n",twt);
 printf("Total TurnAroundTime:%d\n\n",ttat);
 printf("Average Waiting Time:%4.2f\n",avgwt);
 printf("Average TurnAroundTime:%4.2f",avgtat);

}
